cupyx/scipy/ndimage/filters.py

```diff
@@ -720,8 +1133,52 @@ def _generate_boundary_condition_ops(mode, ix, xsize):
     return ops
 
 
+_CAST_FUNCTION = """
+// Implements a casting function to make it compatible with scipy
+// Use like cast<to_type>(value)
+// It's actually really simple - most of this is <type_traits>
```

Comment:
@emcastillo What is being added here is precisely a kind of stubs that jitify would add (#3408), see:
https://github.com/NVIDIA/jitify/blob/2486be5500e06676dd3b2ec8a52dd1a529998fe3/jitify.hpp#L1651

@coderforlife I wonder how did you figure out the minimum stubs needed here to make it work? By trial and error?

Reply:
This looks like a nice, modern C++ approach! It avoids adding an additional argument indicating unsigned status to `_get_correlate_kernel` as would have been required for my simple suggestion in https://github.com/cupy/cupy/issues/3111#issuecomment-650183301

Reply:
@leofang I first wrote it in regular C++ using a regular C++ compiler, making sure it worked, while importing the regular `<type_traits>`. I was quite surprised that `<type_traits>` was not available with NVRTC. I found that CUDA includes this file (in `<cuda/std/type_traits>` which mostly just includes `"cuda/std/detail/libcxx/include/type_traits"`). However, all of those functions would need `__device__` added to them to be usable in NVRTC.

So, I worked backwards. I knew exactly what I needed and looked for them in the CUDA `type_traits` file, combining them with the possible implementations listed on cppreference.com (such as [this](https://en.cppreference.com/w/cpp/types/is_signed#Possible_implementation)) as I thought was appropriate. I did that until I had everything I needed and nothing extra.

Some of them do not quite meet the official standards since I knew I was allowed to cut corners, like knowing every type would always be arithmetic, so skipped that check in `is_signed`.

Reply:
@grlee77 Thanks! I was much happier when I could actually include the C++ header `<type_traits>`... although that might be solved by what leofang is talking about.